### PR TITLE
chore: add a new config `MAX_EXTENDED_MATRIX_SIZE`

### PR DESCRIPTION
**Pull request by kevaundray**
This is pulled out from #3697 

See rationale here: https://github.com/ethereum/consensus-specs/pull/3697#issuecomment-2064924075

Essentially we cannot have a SSZ list where the size is defined a `A * B` where `A` and `B` are constants

### CODE REVIEW

File: specs/_features/eip7594/das-core.md [hunk 1]
```diff
@@ -55,6 +55,7 @@ We define the following Python custom types for type hinting and readability:
 | Name | Value | Description |
 | - | - | - |
 | `NUMBER_OF_COLUMNS` | `uint64(FIELD_ELEMENTS_PER_EXT_BLOB // FIELD_ELEMENTS_PER_CELL)` (= 128) | Number of columns in the extended data matrix. |
+| `MAX_EXTENDED_MATRIX_SIZE` | `uint64(MAX_BLOBS_PER_BLOCK * NUMBER_OF_COLUMNS)` (= 768) | The data size of `ExtendedMatrix`. |
```
Comment (asn-d6) [inline specs/_features/eip7594/das-core.md:58]: IMO `MAX_EXTENDED_MATRIX_SIZE` is a bit ambiguous and gives the impression that this is the maximum number of bytes in an extended matrix, but it's actually the maximum number of cells.

We could  clarify this in its description, or rename the constant to `MAX_CELLS_IN_EXTENDED_MATRIX` or something. Let me know what you think and I can do the change if you want.

**Review by asn-d6 — CHANGES_REQUESTED** [review]
Minor change requested. Otherwise looks good!

File: specs/_features/eip7594/das-core.md (hunk 1 above)
Comment (hwwhww) [reply specs/_features/eip7594/das-core.md:58]: @asn-d6 `MAX_CELLS_IN_EXTENDED_MATRIX` sounds good to me. 👍 feel free to update this PR. Thanks!

Comment (kevaundray) [reply specs/_features/eip7594/das-core.md]: Should be updated now

**Pull request merged**